vault/logical_system_activity_write_testonly.go

```diff
@@ -51,3 +56,133 @@ func (b *SystemBackend) handleActivityWriteData(ctx context.Context, request *lo
 	}
 	return nil, nil
 }
+
+// singleMonthActivityClients holds a single month's client IDs, in the order they were seen
+type singleMonthActivityClients struct {
+	// clients are indexed by ID
+	clients []string
+	// allClients contains all clients from all months
+	allClients map[string]*activity.EntityRecord
+}
+
+// multipleMonthsActivityClients holds multiple month's data
+type multipleMonthsActivityClients struct {
+	// months are in order, with month 0 being the current month and index 1 being 1 month ago
+	months     []*singleMonthActivityClients
+	allClients map[string]*activity.EntityRecord
+}
+
+// addNewClients generates clients according to the given parameters, and adds them to the month
+// if the namespace is empty, the client will be added to the defaultNamespace
+// the client will always have the actualMount as its mount accessor
+func (m *singleMonthActivityClients) addNewClients(c *generation.Client, defaultNamespace, actualMount string) error {
+	count := 1
+	if c.Count > 1 {
+		count = int(c.Count)
+	}
+	for i := 0; i < count; i++ {
+		record := &activity.EntityRecord{
+			ClientID:      c.Id,
+			NamespaceID:   c.Namespace,
+			NonEntity:     c.NonEntity,
+			MountAccessor: actualMount,
+		}
+		if record.ClientID == "" {
+			var err error
+			record.ClientID, err = uuid.GenerateUUID()
+			if err != nil {
+				return err
+			}
+		}
+		if record.NamespaceID == "" {
+			record.NamespaceID = defaultNamespace
+		}
+		m.allClients[record.ClientID] = record
+		seen := 1
+		if c.TimesSeen > 1 {
+			seen = int(c.TimesSeen)
+		}
+		for j := 0; j < seen; j++ {
+			m.clients = append(m.clients, record.ClientID)
+		}
+	}
+	return nil
+}
+
+// processMonth populates a month of client data
+func (m *multipleMonthsActivityClients) processMonth(ctx context.Context, core *Core, month *generation.Data) error {
+	if month.GetAll() == nil {
+		return errors.New("segmented monthly data is not yet supported")
+	}
+
+	// default to using the root namespace and the first mount on the root namespace
+	defaultNamespace := namespace.RootNamespaceID
+	mounts, err := core.ListMounts()
+	if err != nil {
+		return err
+	}
+	defaultMount := ""
+	for _, mount := range mounts {
+		if mount.NamespaceID == defaultNamespace {
+			defaultMount = mount.Accessor
+			break
+		}
+	}
+	if defaultMount == "" {
+		return fmt.Errorf("no mounts found in namespace %s", defaultNamespace)
+	}
+	addingTo := m.months[month.GetMonthsAgo()]
+
+	for _, clients := range month.GetAll().Clients {
+		if clients.Repeated || clients.RepeatedFromMonth > 0 {
+			return errors.New("repeated clients are not yet supported")
+		}
+
+		mountAccessor := defaultMount
+		if clients.Namespace != "" {
+			mountAccessor = ""
```

Comment:
nit: There is a lot of variable overwriting for `mountAccessor`. I wonder if there is a way to avoid that or make it more readable.